.github/workflows/pr.yaml

```diff
@@ -56,30 +56,58 @@ jobs:
 
       - name: Check for outdated bundle
         id: bundle-diff-checker
+        # Lot of debate (https://github.com/janus-idp/operator/pull/195) whether this should be a warning or an error.
+        # For now, this is will be warning + a comment on the PR if manifests are outdated. This way, PR authors/maintainers can be aware of that fact.
+        continue-on-error: true
         run: |
           make bundle
           git status --porcelain
           # Since operator-sdk 1.26.0, `make bundle` changes the `createdAt` field from the bundle every time we run it.
           # The `git diff` below checks if only the createdAt field has changed. If is the only change, it is ignored.
           # Inspired from https://github.com/operator-framework/operator-sdk/issues/6285#issuecomment-1415350333
-          git diff --quiet -I'^    createdAt: ' bundle || ( echo "===================" && \
-            echo "Files changed in bundle generation." && \
-            echo "Please make sure to regenerate the bundle with 'make bundle' and push the changes." && \
-            echo "Make sure you unset any related env vars like VERSION or IMAGE_TAG_BASE or IMG before running this command, as they may affect the resulting manifests." && \
-            echo "For your convenience, the diff will be attached as a job artifact, so you can easily download and Git-apply it right away." && \
-            echo "You might also need to manually update the CSV in '.rhdh/bundle/manifests/rhdh-operator.csv.yaml' file accordingly." && \
-            echo "===================" && \
+          git diff --quiet -I'^    createdAt: ' bundle || ( echo "::group::WARNINGS" && \
             git --no-pager diff | tee bundle.pr-${{ github.event.number }}.patch && \
+            echo "::warning:: Files changed in bundle generation. Please regenerate the bundle with 'make bundle' and push the changes. For your convenience, the diff is attached as a job artifact, so you can easily download and Git-apply it right away instead. You might also need to manually update the CSV in '.rhdh/bundle/manifests/rhdh-operator.csv.yaml' file accordingly." && \
+            echo "::endgroup::" && \
             exit 1 )
 
       - name: Save bundle diff as patch
+        id: bundle-diff-patch-artifact-upload
         uses: actions/upload-artifact@v4
         if: ${{ !cancelled() && steps.bundle-diff-checker.outcome == 'failure' }}
         with:
           name: bundle-diff-patch
           path: bundle.pr-${{ github.event.number }}.patch
           retention-days: 5
 
+      - name: Comment on PR if bundle manifests are outdated
```

Comment:
> it looks like the major part of pr.yaml is dedicated to the case which even does not impact in the end result (it is only about the file which developer generate or does not generate locally)?

Not sure I completely understand your question. You mean a developer making any changes that potentially affect the manifests (even though they are not aware of that in the beginning)?
In that case, I think it should be the responsibility of the CI to make users aware of that fact so that they can fix up their PR. Doing so locally only might be easy to forget or ignore (on purpose or not).

To clarify, we want to prevent the issue where a developer forgets to regenerate the bundle manifests (and potentially update the one in `.rhdh`), causing a mismatch between the code and those manifests.
Regardless of how (error vs warning) we want to make developers aware of such a potential issue, I think it makes sense to enforce this in CI, not only locally.

Reply:
>  we want to make developers aware of such a potential issue, I think it makes sense to enforce this in CI

+1. 

The only way to block the PR from being merged is to fail the GH action if a mismatch occurs. A warning will never be seen by the PR reviewer.